examples/liver-scriptcontroller.py

```diff
@@ -0,0 +1,103 @@
+    root.addObject('DefaultPipeline', name="CollisionPipeline")
+    root.addObject('BruteForceDetection', name="N2")
+    root.addObject('DefaultContactManager', name="CollisionResponse", response="default")
+    root.addObject('DiscreteIntersection')
+
+    root.addObject('MeshObjLoader', name="LiverSurface", filename="mesh/liver-smooth.obj")
+
+    liver = root.addChild('Liver')
+    liver.addObject('EulerImplicitSolver', name="cg_odesolver", rayleighStiffness="0.1", rayleighMass="0.1")
+    liver.addObject('CGLinearSolver', name="linear_solver", iterations="25", tolerance="1e-09", threshold="1e-09")
+    liver.addObject('MeshGmshLoader', name="meshLoader", filename="mesh/liver.msh")
+    liver.addObject('TetrahedronSetTopologyContainer', name="topo", src="@meshLoader")
+    liver.addObject('MechanicalObject', name="dofs", src="@meshLoader")
+    liver.addObject('TetrahedronSetGeometryAlgorithms', template="Vec3d", name="GeomAlgo")
+    liver.addObject('DiagonalMass', name="Mass", massDensity="1.0")
+    liver.addObject('TetrahedralCorotationalFEMForceField', template="Vec3d", name="FEM", method="large", poissonRatio="0.3", youngModulus="3000", computeGlobalMatrix="0")
+    liver.addObject('FixedConstraint', name="FixedConstraint", indices="3 39 64")
+
+    visu = liver.addChild('Visu')
+    visu.addObject('OglModel', name="VisualModel", src="@../../LiverSurface")
+    visu.addObject('BarycentricMapping', name="VisualMapping", input="@../dofs", output="@VisualModel")
+
+    surf = liver.addChild('Surf')
+    surf.addObject('SphereLoader', name="sphereLoader", filename="mesh/liver.sph")
+    surf.addObject('MechanicalObject', name="spheres", position="@sphereLoader.position")
+    surf.addObject('SphereCollisionModel', name="CollisionModel", listRadius="@sphereLoader.listRadius")
+    surf.addObject('BarycentricMapping', name="CollisionMapping", input="@../dofs", output="@spheres")
+
+    root.addObject(KeyPressedController(name = "SphereCreator"))
+
+    return root
+
+
+class KeyPressedController(Sofa.Core.Controller):
+    """ This controller monitors new sphere objects.
+    Press ctrl and the L key to make spheres falling!
+    """
+    def __init__(self, *args, **kwargs):
+        Sofa.Core.Controller.__init__(self, *args, **kwargs)
+        self.listening = True
+        self.name = "keyPressedController"
+        self.iteration = 0
+
+    def onKeypressedEvent(self, event):
+        # Press L key triggers the creation of new objects in the scene
+        if event['key']=='L':
+            self.createNewSphere()
+            
+    def createNewSphere(self):
+        root = self.getContext()
+        newSphere = root.addChild('FallingSphere-'+str(self.iteration))
+        newSphere.addObject('EulerImplicitSolver')
+        newSphere.addObject('CGLinearSolver', threshold='1e-09', tolerance='1e-09', iterations='200')
```

Comment:
No need for string argument everywhere

```suggestion
        newSphere.addObject('CGLinearSolver', threshold=1e-09, tolerance=1e-09, iterations=200)
```